core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -446,6 +447,7 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                 transportService, indicesService, pluginsService, circuitBreakerService, scriptModule.getScriptService(),
                 httpServerTransport, ingestService, clusterService, settingsModule.getSettingsFilter());
             modules.add(b -> {
+                    b.bind(ResponseCollectorService.class).toInstance(responseCollectorService);
```

Comment:
I don't think we should do this! Let's please don't add any guice related stuff anymore no matter what it costs.